File: examples/typescript/client_websocket_ttl.ts
/**
 * WebSocket TTL - Using @ekodb/ekodb-client library
 */

import { EkoDBClient } from "@ekodb/ekodb-client";
import * as dotenv from "dotenv";

dotenv.config();

const BASE_URL = process.env.API_BASE_URL || "http://localhost:8080";
const WS_URL = process.env.WS_BASE_URL || "ws://localhost:8080";
const API_KEY = process.env.API_BASE_KEY || "a-test-api-key-from-ekodb";

async function main() {
  const client = new EkoDBClient(BASE_URL, API_KEY);
  await client.init();
  console.log("✓ Client created");

  const collection = "client_websocket_ttl_ts";

  console.log("\n=== Insert Test Data with TTL ===");
  const record = { name: "WebSocket TTL Test", value: 42, active: true };
  const inserted = await client.insert(collection, record, "1h");
  console.log(`✓ Inserted document with TTL: ${inserted.id}`);

  console.log("\n=== Query via WebSocket ===");
  const ws = client.websocket(WS_URL);
  console.log("✓ WebSocket connected");

  const records = await ws.findAll(collection);
  console.log(`✓ Retrieved ${records.length} record(s) via WebSocket`);

  for (let i = 0; i < Math.min(3, records.length); i++) {
    const fieldCount = Object.keys(records[i]).length;
    console.log(`  Record ${i + 1}: ${fieldCount} fields`);
  }

  ws.close();

  // Cleanup: Delete the collection
  console.log("\n=== Cleanup ===");
  await client.deleteCollection(collection);
  console.log("✓ Deleted collection");

  console.log("\n✓ WebSocket TTL example completed successfully");
  console.log(
    "\n💡 Note: Documents with TTL will automatically expire after the specified duration",
  );
}

main().catch(console.error);

File: examples/python/client_websocket_ttl.py
#!/usr/bin/env python3
"""WebSocket TTL Examples - Using ekodb_client Python library

This example demonstrates WebSocket queries with TTL documents using the Python client library.
Compare with ttl-caching/websocket_ttl.py to see the difference!
"""

import asyncio
import os
from pathlib import Path
from dotenv import load_dotenv

# Import the ekodb_client (Python bindings for Rust client)
from ekodb_client import Client

# Load environment variables
env_path = Path(__file__).parent.parent / ".env"
load_dotenv(env_path)

BASE_URL = os.getenv("API_BASE_URL", "http://localhost:8080")
WS_URL = os.getenv("WS_BASE_URL", "ws://localhost:8080")
API_KEY = os.getenv("API_BASE_KEY", "a-test-api-key-from-ekodb")


async def websocket_ttl_examples():
    """Run WebSocket TTL examples using ekodb_client"""
    # Create ekoDB client
    client = Client.new(BASE_URL, API_KEY)
    print("✓ Client created")

    collection = "client_websocket_ttl_python"

    try:
        # Step 1: Insert test data with TTL
        print("\n=== Insert Test Data with TTL ===")
        doc = await client.insert(
            collection,
            {"name": "WebSocket TTL Test", "value": 42, "active": True},
            ttl="1h",
        )
        doc_id = doc.get("id")
        print(f"✓ Inserted document with TTL: {doc_id}")

        # Step 2: Query via WebSocket
        print("\n=== Query via WebSocket ===")
        ws_client = await client.websocket(WS_URL)
        print("✓ WebSocket connected")

        records = await ws_client.find_all(collection)
        print(f"✓ Retrieved {len(records)} record(s) via WebSocket")

        # Show some of the data
        for i, record in enumerate(records[:3], 1):
            field_count = len(record) if isinstance(record, dict) else 0
            print(f"  Record {i}: {field_count} fields")

        # Cleanup: Delete the collection
        print("\n=== Cleanup ===")
        await client.delete_collection(collection)
        print("✓ Deleted collection")

        print("\n✓ WebSocket TTL example completed successfully")
        print(
            "\n💡 Note: Documents with TTL will automatically expire after the specified duration"
        )

    except Exception as e:
        print(f"Error: {e}")
        raise


if __name__ == "__main__":
    asyncio.run(websocket_ttl_examples())
